### DISCUSSION TITLE
How to make whole window direction right to left?

### DISCUSSION THREAD

**Discussion by khaledmahmoud78** [picoe/Eto#2691] [Q&A]
I am an Arab, I would like to make apps in Arabic language.
So I need to switch the direction of the window and all controls  to RTL.
In WinForms, There is a property named "RightToLeft" to switch control direction.
How I can do this in Eto.Forms?
Thanks in advance!

**Comment by cwensley** [comment]
Hi @khaledmahmoud78 ,

Thanks for the inquiry. Unfortunately no RTL support has been added to Eto at this point. 

I would definitely welcome supporting this though. I'm just not sure how difficult it would be to add to each of the platforms.

**Comment by Serg-Norseman** [comment]
A pressing question. A month ago, a user of my program also asked about adding RTL support for Arabic. I couldn't help him.

**Comment by ghost** [comment]
In Eto.Forms, to switch the direction of the window and all controls to Right-to-Left (RTL) for Arabic or similar languages, you can set the FlowDirection property of the controls and form to RightToLeft.

Here's how you can achieve this:

For the Form itself, set the FlowDirection property to RightToLeft:
`var form = new Form
{
    Title = "Arabic App",
    ClientSize = new Eto.Drawing.Size(400, 300),
    FlowDirection = Eto.Drawing.FlowDirection.RightToLeft
};
`
For individual controls (e.g., buttons, labels), you can also set the FlowDirection property to RightToLeft to ensure their contents align correctly:
` var label = new Label { Text = "مرحبا بكم" };
label.FlowDirection = Eto.Drawing.FlowDirection.RightToLeft;

var button = new Button { Text = "اضغط هنا" };
button.FlowDirection = Eto.Drawing.FlowDirection.RightToLeft;
`
If you want the entire application to support RTL, you can iterate over all controls on the form and set their FlowDirection property in a similar way.
This will ensure that both the form and the controls are aligned properly for RTL languages like Arabic.

**Reply by cwensley** [reply]
Hi @Vikas-Devnet,

This is incorrect, and it seems to me that you simply used ChatGPT/Copilot to generate this answer.  There is currently no `FlowDirection` property of either `Label` or `Form`, but that would certainly be one way it _would_ be implemented.
